### PR TITLE
DPL: enable shared memory backend

### PR DESCRIPTION
**Pull request by ktf**
Devices sitting on the same resource will use the same shared memory
pool. Notice how we set the FairMQ `--session` to be
`dpl_<driver-process-pid>`.

### CODE REVIEW

**Comment by ktf** [comment]
@aalkin This will hopefully allow multiple workflows to run on the same node, using different shared memory pools.

**Comment by aalkin** [comment]
@ktf test_ProcessorOptions seems to have same faulty quit logic as test_DanglingOutputs had. RootTreeWriterWorkflow fails only on linux, which I can reproduce locally, it gets stuck in processDanglingInputs.

**Comment by ktf** [comment]
ok, if you manage to fix it just do a pr to my branch and i will merge it.

**Comment by ktf** [comment]
or even feel free to open a new one

File: Framework/Core/src/DataProcessingDevice.cxx [hunk 1]
```diff
@@ -239,6 +239,9 @@ bool DataProcessingDevice::ConditionalRun()
     if (info.state != InputChannelState::Running) {
       continue;
     }
+    if (mState.quitRequested) {
```
Comment (aalkin) [inline Framework/Core/src/DataProcessingDevice.cxx]: This change is what breaking RootTreeWriterWorkflow: it relies on sink1 to keep forwarding messages from the source to sink2, which sink1 is not doing anymore as it skips incoming inputs entirely. This means we need to rework the enofstream/quit mechanism...

**Comment by aalkin** [comment]
It seems the devices are not actually using shared memory, judging by the failure logs, printed configs at the start have zeromq set as transport. It seems that to use non-default transport one needs to explicitly override it with `--transport shmem` which will only affect channels where the custom transport is set (i.e. with `,transport=shmem` in --channel-config).

**Comment by ktf** [comment]
There is still something wrong. This leaves behind a bunch of fairmq-shmmonit zombies on quit. Reverting.

@rbx @dennisklein any idea of what might be causing it?

**Pull request merged**